File: app/src/main/java/org/magnum/logger/OneTimer.java
package org.magnum.logger;

import android.app.AlarmManager;
import android.app.PendingIntent;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.location.LocationManager;
import android.os.IBinder;
import android.util.Log;

import org.magnum.logger.user.applications.RunningApplicationHandler;
import org.magnum.logger.user.location.LocationHandler;

public class OneTimer extends Service
{
	private static final String TAG = OneTimer.class.getCanonicalName();

	@Override
	public int onStartCommand(Intent intent, int flags, int startId)
	{
		try
		{
			monitorLocation();
			monitorApps();
		}
		catch (Exception e)
		{
			Log.e(TAG, e.toString());
		}
		return START_NOT_STICKY;
	}



	private void monitorLocation()
	{
		LocationManager manager = (LocationManager) this.getSystemService(Context.LOCATION_SERVICE);
		if (manager != null)
		{
			manager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0, 1610, new LocationHandler(getApplicationContext()));
		}
	}

	private void monitorApps()
	{
		PendingIntent pendingIntent = PendingIntent.getService(getApplicationContext(), 0, new Intent(this, RunningApplicationHandler.class), 0);
		AlarmManager alarmManager = (AlarmManager) getSystemService(ALARM_SERVICE);
		if (alarmManager != null)
		{
			alarmManager.setRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), AlarmManager.INTERVAL_FIFTEEN_MINUTES / 15, pendingIntent);
		}
	}

	@Override
	public IBinder onBind(Intent intent)
	{
		return null;
	}

}

File: app/src/main/java/org/magnum/logger/OneTimer.kt
package org.magnum.logger

import android.Manifest
import android.app.AlarmManager
import android.app.PendingIntent
import android.app.Service
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.location.LocationManager
import android.os.IBinder
import android.util.Log
import androidx.core.app.ActivityCompat
import org.magnum.logger.user.applications.RunningApplicationHandler
import org.magnum.logger.user.location.LocationHandler

class OneTimer : Service() {
    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        try {
            monitorLocation()
            monitorApps()
        } catch (e: Exception) {
            Log.e(TAG, e.toString())
        }
        return START_NOT_STICKY
    }

    private fun monitorLocation() {
        val manager = this.getSystemService(Context.LOCATION_SERVICE) as LocationManager
        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
            manager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0, 1610f, LocationHandler(applicationContext))
        }
    }

    private fun monitorApps() {
        val pendingIntent = PendingIntent.getService(applicationContext, 0, Intent(this, RunningApplicationHandler::class.java), 0)
        val alarmManager = getSystemService(Context.ALARM_SERVICE) as AlarmManager
        alarmManager.setRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), AlarmManager.INTERVAL_FIFTEEN_MINUTES / 15, pendingIntent)
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    companion object {
        private val TAG = OneTimer::class.java.canonicalName
    }
}